sql/sql_table.cc

```diff
@@ -9807,17 +9809,42 @@ bool mysql_checksum_table(THD *thd, TABLE_LIST *tables,
                 case MYSQL_TYPE_GEOMETRY:
                 case MYSQL_TYPE_BIT:
                 {
+                  if (checksum_start)
+                  {
+                    row_crc= my_checksum(row_crc, checksum_start, checksum_length);
+                    checksum_start= NULL;
+                    checksum_length= 0;
+                  }
                   String tmp;
                   f->val_str(&tmp);
                   row_crc= my_checksum(row_crc, (uchar*) tmp.ptr(),
                            tmp.length());
                   break;
                 }
                 default:
-                  row_crc= my_checksum(row_crc, f->ptr, f->pack_length());
+                  if (checksum_start)
+                  {
+                    if (checksum_start + checksum_length == f->ptr)
```

Comment:
Question to MariaDB devs: can this condition be false? If not this whole diff hunk can be replaced with something like:
if (!checksum_start)
  checksum_start= f->ptr;
DBUG_ASSERT(checksum_start + checksum_length == f->ptr);
checksum_length+= f->pack_length();